### PR TITLE
docs: add digital ocean page

### PR DESCRIPTION
**Pull request by danielbarion**
- Add one digital ocean article to the website to thank them for their support in open source projects.

## Summary by CodeRabbit

- **New Features**
	- Added a new footer link titled "Deploying to DigitalOcean" for easier access to deployment documentation.
	- Introduced a guide for deploying a static site on DigitalOcean's App Platform, featuring step-by-step instructions and visual aids.

- **Documentation**
	- Enhanced documentation structure with new content on deploying React applications.
  
- **Chores**
	- Added TypeScript declarations for various image file formats to improve development experience.

### CODE REVIEW

File: docs/src/pages/blog/digital-ocean-app-platform.tsx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+import DigitalOceanLogin from '@site/static/img/digital-ocean-login.webp'
+// @ts-ignore
+import DigitalOceanCreateApp from '@site/static/img/digital-ocean-create-app.webp'
+// @ts-ignore
+import DigitalOceanAppConfig1 from '@site/static/img/digital-ocean-app-config-1.png'
+// @ts-ignore
+import DigitalOceanAppConfig2 from '@site/static/img/digital-ocean-app-config-2.png'
+// @ts-ignore
+import DigitalOceanStaticSite from '@site/static/img/digital-ocean-static-site.png'
+// @ts-ignore
```
Comment (gabrieljablonski) [inline docs/src/pages/blog/digital-ocean-app-platform.tsx:17]: Try adding a `react-app-env.d.ts` to the `docs/` directory. With that, we can avoid these `@ts-ignore`.

```ts
declare module '*.png'
declare module '*.svg'
declare module '*.jpeg'
declare module '*.jpg'
declare module '*.webp'
```

**Review by gabrieljablonski — APPROVED** [review]
LGTM (with one comment about `@ts-ignore`), but I'd be fine with making the placement a *bit* more obvious, other than just on the page footer.

Maybe close to the "powered by" text? Your call if needed.

File: docs/src/pages/blog/digital-ocean-app-platform.tsx (hunk 1 above)
Comment (danielbarion) [reply docs/src/pages/blog/digital-ocean-app-platform.tsx:17]: Done, thanks!

Comment (danielbarion) [reply docs/src/pages/blog/digital-ocean-app-platform.tsx]: image
image

**Review by gabrieljablonski — APPROVED** [review]
LGTM

**Pull request merged**